lib/bankaccount/checking_account.rb

```diff
@@ -0,0 +1,59 @@
+module BankAccount
+
+	class CheckingAccount < Account
+
+		attr_reader :id, :initial_balance, :account_balance, :free_checks
+
+		def initialize(id, initial_balance)
+			raise ArgumentError.new "Please enter a positive initial balance." unless initial_balance > 0
```

Comment:
I like this line. I know you've been taught to do ArgumentErrors but I hadn't at your point in the program so they all seem super shiny and advanced. Also using the `unless` afterward is super slick!